### PR TITLE
Allow reordering by default when reading from file/s3/url

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Changed default values of settings parallelize_output_from_storages and input_format_parquet_preserve_order. This allows ClickHouse to reorder rows when reading from files (e.g. CSV or Parquet), greatly improving performance in many cases. To restore the old behavior of preserving order, use `parallelize_output_from_storages = 0`, `input_format_parquet_preserve_order = 1`.

Just change defaults for the two settings.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 927b6834dba577e48d908a843e9b35ee04fc8fd9 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49479/927b6834dba577e48d908a843e9b35ee04fc8fd9/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/00900_long_parquet_load.sh [hunk 1]
```diff
@@ -66,6 +66,6 @@ EOF
     # Some files contain unsupported data structures, exception is ok.
     cat "$DATA_DIR"/"$NAME" | ${CLICKHOUSE_CLIENT} --query="INSERT INTO parquet_load FORMAT Parquet" 2>&1 | sed 's/Exception/Ex---tion/'
 
-    ${CLICKHOUSE_CLIENT} --query="SELECT * FROM parquet_load LIMIT 100"
+    ${CLICKHOUSE_CLIENT} --query="SELECT * FROM parquet_load ORDER BY tuple(*) LIMIT 100"
```
Comment (qoega) [inline tests/queries/0_stateless/00900_long_parquet_load.sh:69]: Why do we need it? Just curious

**Comment by qoega** [comment]
Do we want it to be added as to compatibility settings as well? Or we believe this is rare case to depend on file read order and better to enable it for all?

**Comment by alexey-milovidov** [comment]
Yes, SettingsChangesHistory plz.

File: tests/queries/0_stateless/00900_long_parquet_load.sh (hunk 1 above)
Comment (al13n321) [reply tests/queries/0_stateless/00900_long_parquet_load.sh:69]: The order of the rows is not deterministic anymore.

Comment (qoega) [reply tests/queries/0_stateless/00900_long_parquet_load.sh:69]: I misread `tuple()` instead of `tuple(*)` and thought how it helps. Thanks

**Comment by al13n321** [comment]
Failing tests:
 * 02723_zookeeper_name - broken,
 * 02725_agg_projection_resprect_PK - broken,
 * 02479_mysql_connect_to_self - says "Lost connection to MySQL server during query", so I guess it's just flaky?
 * Stress test says:
```
E               Exception: Command ['docker', 'exec', '-u', 'root', 'roottestdiskoverwebserver_node1_1', 'bash', '-c', '/usr/bin/clickhouse static-files-disk-uploader --test-mode --url http://nginx:80/test1 --metadata-path /store/d33/d33e5d97-617b-4afa-9f74-79bc4246d7a8/'] return non-zero code 1: Data path: "/store/d33/d33e5d97-617b-4afa-9f74-79bc4246d7a8", destination path: "http://nginx:80/test1"
E               std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory ["/store/d33/d33e5d97-617b-4afa-9f74-79bc4246d7a8/tmp_merge_all_1_10_2"]
```
sounds unrelated.

File: src/Core/SettingsChangesHistory.h [hunk 2]
```diff
@@ -80,8 +80,10 @@ namespace SettingsChangesHistory
 /// It's used to implement `compatibility` setting (see https://github.com/ClickHouse/ClickHouse/issues/35972)
 static std::map<ClickHouseVersion, SettingsChangesHistory::SettingsChanges> settings_changes_history =
 {
-    {"23.4", {{"allow_suspicious_indices", true, false, "If true, index can defined with identical expressions"}}},
-    {"23.4", {{"connect_timeout_with_failover_ms", 50, 1000, "Increase default connect timeout because of async connect"},
+    {"23.5", {{"input_format_parquet_preserve_order", true, false, "Allow Parquet reade to reorder rows for better parallelism."},
```
Comment (alexey-milovidov) [inline src/Core/SettingsChangesHistory.h:83]: `reade` - typo

Comment (al13n321) [reply src/Core/SettingsChangesHistory.h:83]: Oops. I'll piggy-back the fix to some unrelated PR. (Unless it's important enough for its own PR, or we have some different process for this, or sth.)

**Pull request merged**